### PR TITLE
[SYCL] Detect conflicts between kernel properties

### PR DESCRIPTION
**Pull request by frasercrmck**
The `max_work_group_size` and `max_linear_work_group_size` kernel properties conflict with the `work_group_size` property when the required work-group size exceeds either of the maximum sizes.

### CODE REVIEW

File: sycl/include/sycl/ext/oneapi/kernel_properties/properties.hpp [hunk 1]
```diff
@@ -351,6 +350,78 @@ struct HasKernelPropertiesGetMethod<T,
...
+// trivially exceed the maximum size.
+template <typename Properties>
+struct ConflictingProperties<max_work_group_size_key, Properties>
+    : std::false_type {
+  using WGSizeVal = GetWGPropertyFromPropList<work_group_size_key, Properties>;
+  using MaxWGSizeVal =
+      GetWGPropertyFromPropList<max_work_group_size_key, Properties>;
+  static constexpr size_t Dims = WGSizeVal::WGSize.size();
+  static_assert(
+      Dims == 0 || Dims == MaxWGSizeVal::WGSize.size(),
```
Comment (steffenlarsen) [inline sycl/include/sycl/ext/oneapi/kernel_properties/properties.hpp:401]: It took me a moment to realize that `Dims == 0` is for the case where `WGSizeVal` doesn't exist in the list. Definitely clever, but would you mind adding a comment to make it a little easier for the next time someone comes by this?

**Review by steffenlarsen — APPROVED** [review]
Very clever! Small nit, but otherwise I think it looks good!

File: sycl/include/sycl/ext/oneapi/kernel_properties/properties.hpp (hunk 1 above)
Comment (frasercrmck) [reply sycl/include/sycl/ext/oneapi/kernel_properties/properties.hpp:403]: Yeah, good idea. Even took me a while to remember what I'd done.

**Comment by frasercrmck** [comment]
@intel/llvm-gatekeepers this is ready to merge, thank you!

**Pull request merged**